object_storage_api/repositories/image.py

```diff
@@ -49,13 +50,21 @@ def get(self, image_id: str, session: ClientSession = None) -> Optional[ImageOut
         :param image_id: ID of the image to retrieve.
         :param session: PyMongo ClientSession to use for database operations.
         :return: Retrieved image or `None` if not found.
+        :raises MissingRecordError: If the supplied `image_id` is invalid or non-existent.
         """
-        image_id = CustomObjectId(image_id)
         logger.info("Retrieving image with ID: %s from the database", image_id)
-        image = self._images_collection.find_one({"_id": image_id}, session=session)
+        message = f"Image with image_id {image_id} was not found."
+        entity_name = "image"
+        try:
+            image_id = CustomObjectId(image_id)
+            image = self._images_collection.find_one({"_id": image_id}, session=session)
+        except InvalidObjectIdError as e:
+            exc = MissingRecordError(detail=message, entity_name=entity_name)
+            raise exc from e
```

Comment:
```suggestion
        except InvalidObjectIdError as exc:
            raise MissingRecordError(detail=message, entity_name=entity_name) from exc
```
This and the one below can go on one line now.